fedoracommunity/consumers.py

```diff
@@ -146,28 +142,18 @@ def clear_and_refresh_items(filters):
 
     def update_xapian(self, msg):
 
-        # TODO - this needs to be rethought to figure out *when* to update the
-        # fedora-packages xapian cache.  It used to happen anytime anything
-        # changed in pkgdb, but when is a new good event?  Yash - you don't
-        # have to modify this function (in the beginning).  Save it for last.
-
-        # If any number of different pkgdb things happen to a package, let's
-        # just update and not care too much about whatever it was that just
-        # happened.
-        if '.pkgdb.' not in msg['topic']:
-            return
-
-        # This one is spammy
-        if '.pkgdb.acl.update' in msg['topic']:
+        # If any repo update things happen to packages in mdapi then
+        # just update the cache.
+        
+        #filter for mdapi msgs
+        if '.mdapi.repo.update' not in msg['topic']:
             return
 
         # We'll take all others, so long as they have these fields.
-        if 'package_listing' in msg['msg']:
-            name = msg['msg']['package_listing']['package']['name']
-        elif 'package' in msg['msg']:
-            name = msg['msg']['package']['name']
+        if 'packages' in msg['msg']:
+            name = msg['msg']['name']
         else:
-            # If the message doesn't have either of those two, then give up.
+            # If the message doesn't have a name, then give up.
             return
 
         log.info("Considering xapian index updates for %r" % name)
```

Comment:
The logic down here that updates the xapian index...

It previously was run *once* for a single *pkgdb* message.  pkgdb messages were about *single* packages.

.. but these mdapi messages are about *multiple* packages.  This code here, and below, will need to be indented to be inside a for loop.  It'll be run multiple times per mdapi message -- once for every package listed inside the message.